core/txpool/legacypool/legacypool.go

```diff
@@ -568,9 +572,19 @@ func (pool *LegacyPool) Pending(filter txpool.PendingFilter) map[common.Address]
 				}
 			}
 		}
+		if filter.MaxDATxSize != nil && !pool.locals.contains(addr) {
+			for i, tx := range txs {
+				if types.EstimatedL1SizeScaled(tx.RollupCostData()).Cmp(maxDASizeScaled) > 0 {
+					txs = txs[:i]
+					break
+				}
+			}
+		}
 		if len(txs) > 0 {
 			lazies := make([]*txpool.LazyTransaction, len(txs))
 			for i := 0; i < len(txs); i++ {
+				daBytes := types.EstimatedL1SizeScaled(txs[i].RollupCostData())
+				daBytes.Div(daBytes, big.NewInt(1e6))
```

Comment:
Nit: `daBytes = daBytes.Div(daBytes, big.NewInt(1e6))` is slightly more readable, and requires less knowledge of the `big` package's mutation behavior.

Reply:
changed